esprima.js

```diff
@@ -3034,13 +3042,14 @@
             }
             return {
                 type: PlaceHolders.ArrowParameterPlaceHolder,
-                params: []
+                params: [],
+                rawParams: []
             };
         }
 
         startToken = lookahead;
         if (match('...')) {
-            expr = parseRestElement();
+            expr = parseRestElement(params);
```

Comment:
`params` is an array to receive all the bound names. Normally bound name duplication is not allowed anywhere, we are not checking them yet.